viewer_clients/web-client/scripts/codecheckerviewer/BugViewer.js

```diff
@@ -462,49 +462,204 @@ function (declare, dom, style, on, query, Memory, Observable, topic, entities,
+        var contains = function(substr) {
+          return msg.indexOf(substr) !== -1;
+        };
+        var startsWith = function(substr) {
+          return msg.startsWith(substr);
+        };
+        var extractFuncName = function(prefix) {
+          if (startsWith(prefix)) {
+            return msg.replace(prefix, "").replace("'", "");
+          }
+          return false;
+        };
+
+        var func;
+        if (func = extractFuncName("Calling ")) {
+          stack.funcStack.push(func);
+          stack.background = that._highlight_colours[stack.funcStack.length % that._highlight_colours.length];
+
+          highlight.iconOverride = "calling";
+        } else if (startsWith("Entered call from ")) {
+          highlight.iconOverride = "entered_call";
+        } else if (func = extractFuncName("Returning from ")) {
+          if (func === stack.funcStack[stack.funcStack.length - 1]) {
+            stack.funcStack.pop();
+            stack.background = that._highlight_colours[stack.funcStack.length % that._highlight_colours.length];
+
+            highlight.iconOverride = "returning";
+          } else {
+            throw {
+              name : 'StackError',
+              message : "Returned from " + func + " while the last function was " +
+                        stack.funcStack[stack.funcStack.length - 1]
+            };
+          }
+        } else if (startsWith("Assuming the condition")) {
+          highlight.iconOverride = "assume_switch";
+        } else if (startsWith("Assuming")) {
+          highlight.iconOverride = "assume_exclamation";
+        } else if (msg == "Entering loop body") {
+          highlight.iconOverride = "loop_enter";
+        } else if (startsWith("Loop body executed")) {
+          highlight.iconOverride = "loop_execute";
+        } else if (msg == "Looping back to the head of the loop") {
+          highlight.iconOverride = "loop_back";
+        }
+      };
+
+      stackTransition();
+
+      return highlight;
+    },
+
     _addReport : function (report) {
       var that = this;
 
       this.bugStore.put({
         id : report.reportId + '',
         name : 'Line ' + report.lastBugPosition.startLine
-             + ': ' + report.checkerId,
+             + ' &ndash; ' + report.checkerId,
         parent : util.severityFromCodeToString(report.severity),
         report : report,
-        isLeaf : false
+        isLeaf : false,
+        kind : 'bugpath'
       });
 
       this.bugStore.put({
         id : report.reportId + '_0',
-        name : '<b><u>Result</u>: '
-             + entities.encode(report.checkerMsg)
-             + '</b>',
+        name : '<b><u>' + entities.encode(report.checkerMsg) + '</u></b>',
         parent : report.reportId + '',
         report : report,
-        isLeaf : true
+        isLeaf : true,
+        kind : 'msg'
       });
 
+      var filePaths = [];
+      var areThereMultipleFiles = false;
+      var highlightStack = {
```

Comment:
Reduce the scope of these variables. These are used only in the callback of `getReportDetails`.